oauth2/oauth2-client/src/main/java/org/springframework/security/oauth2/client/oidc/authentication/OidcIdTokenValidator.java

```diff
@@ -146,8 +146,8 @@ public void setClock(Clock clock) {
 		this.clock = clock;
 	}
 
-	private static OAuth2Error invalidIdToken(Map<String, Object> invalidClaims) {
-		return new OAuth2Error("invalid_id_token", "The ID Token contains invalid claims: " + invalidClaims,
+	private static OAuth2InvalidTokenError invalidIdToken(Map<String, Object> invalidClaims) {
```

Comment:
Since client error codes aren't specified in the OIDC spec, I think it's fine to have this be `invalid_id_token`.

What are your reasons for wanting to change OIDC client error reporting?

Reply:
This was a missunderstanding on my side, I will revert that change.